### PR TITLE
Organisation Schema and DELETE request added

### PR DESCRIPTION
**Pull request by tanay13**
I created a new folder for organisation schema and middleware to check the ownership of the organisation .

### CODE REVIEW

**Comment by tanay13** [comment]
@YashKumarVerma  please check the approach

File: src/modules/user/routes.ts [hunk 1]
```diff
@@ -54,4 +55,16 @@ router.get('/login',  (req, res) => {
...
   
+  // Delete organisation
+  router.delete('/:orgusername',OrgOwnership,(req:any,res:any)=>{
+    org.deleteOne({orgusername:req.params.orgusername},(err:any)=>{
+      if(err){
+        res.redirect('/home');
+      }else{
+        res.redirect('/home');
+      }
+    })
```
Comment (YashKumarVerma) [inline src/modules/user/routes.ts:66]: would work, how about moving it to a controller file ?

**Comment by tanay13** [comment]
@YashKumarVerma please check the code , i have made changes .
I commented out middleware code as it was giving errors , i will resolve that soon .

File: src/modules/organization/controllers.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { Request, Response } from 'express'
+import { Request, Response, response } from 'express'
```
Comment (YashKumarVerma) [inline src/modules/organization/controllers.ts:1]: response is not needed. 
Only two are enough to work.

File: src/modules/organization/controllers.ts [hunk 3]
```diff
@@ -21,6 +21,19 @@ class OrganizationOperations {
...
+      response.json(data)
+    }
+  catch(err){
+    response.json({
+      error:err.error,
+      message:err.message
+    })
+  }
+ }
+
```
Comment (YashKumarVerma) [inline src/modules/organization/controllers.ts:36]: good !

File: src/modules/organization/models.ts [hunk 4]
```diff
@@ -56,6 +59,29 @@ class UserModel {
         })
     })
   }
+ 
+  static async deleteOrg(organization:organizationDatatype,req:express.Request): Promise<returnDataType>{
+    return new Promise((resolve,reject)=>{
```
Comment (YashKumarVerma) [inline src/modules/organization/models.ts:64]: import express datatypes (express.request ...) like done in controller.

File: src/modules/organization/models.ts [hunk 5]
```diff
@@ -56,6 +59,29 @@ class UserModel {
...
+     Organization.deleteOne({name:req.params.orgusername})
```
Comment (YashKumarVerma) [inline src/modules/organization/models.ts:65]: delete by id, not name. name can be same too

File: src/modules/organization/routes.ts [hunk 6]
```diff
@@ -31,6 +31,5 @@ router.post('/register', Controller.register)
 
 /** Export router to be included into main application */
 
-// Login handle
-
+router.delete('/:orgusername',Controller.delete)
```
Comment (YashKumarVerma) [inline src/modules/organization/routes.ts:34]: take id here

**Comment by tanay13** [comment]
@YashKumarVerma done..

**Comment by YashKumarVerma** [comment]
Good work !

**Comment by YashKumarVerma** [comment]
Please close the issue mentioning this pull request. and from next PRs, use something like fixes #issueNumber

**Pull request merged**